packages/backend/src/managers/llama-stack/llamaStackManager.spec.ts

```diff
@@ -0,0 +1,330 @@
+  vi.mocked(podmanConnection.onPodmanConnectionEvent).mockImplementation(f => {
+    f({
+      status: 'started',
+    });
+    return NO_OP_DISPOSABLE;
+  });
+
+  llamaStackManager.init();
+
+  expect(llamaStackManager.refreshLlamaStackContainer).toHaveBeenCalledWith();
+
+  await vi.waitFor(() => {
+    expect(llamaStackManager.getContainerInfo()).toEqual({
+      containerId: 'dummyId',
+      port: 50000,
+    });
+  });
+});
+
+test('onPodmanConnectionEvent stop event should call refreshLlamaStackContainer and clear containerInfo', async () => {
+  vi.spyOn(llamaStackManager, 'refreshLlamaStackContainer');
+  vi.mocked(containerEngine.listContainers).mockResolvedValueOnce([LLAMA_STACK_CONTAINER_RUNNING]);
+  vi.mocked(podmanConnection.onPodmanConnectionEvent).mockImplementation(f => {
+    f({
+      status: 'started',
+    });
+    setTimeout(() => {
+      f({
+        status: 'stopped',
+      });
+    }, 100);
+    return NO_OP_DISPOSABLE;
+  });
+
+  llamaStackManager.init();
+
+  expect(llamaStackManager.refreshLlamaStackContainer).toHaveBeenCalledWith();
+
+  await vi.waitFor(() => {
+    expect(llamaStackManager.getContainerInfo()).toEqual({
+      containerId: 'dummyId',
+      port: 50000,
+    });
+  });
+
+  // Prepare to receive new event in ~ 100ms
+
+  vi.mocked(llamaStackManager.refreshLlamaStackContainer).mockClear();
+  vi.mocked(containerEngine.listContainers).mockResolvedValueOnce([LLAMA_STACK_CONTAINER_STOPPED]);
+
+  // shoudl happen after 100ms
+  await vi.waitFor(async () => {
+    expect(llamaStackManager.getContainerInfo()).toBeUndefined();
+  });
+  expect(llamaStackManager.refreshLlamaStackContainer).toHaveBeenCalledWith();
+});
+
+test('onStartContainerEvent event should call refreshLlamaStackContainer and set containerInfo', async () => {
+  vi.spyOn(llamaStackManager, 'refreshLlamaStackContainer');
+  vi.mocked(containerEngine.listContainers).mockResolvedValueOnce([LLAMA_STACK_CONTAINER_RUNNING]);
+  vi.mocked(containerRegistry.onStartContainerEvent).mockImplementation(f => {
+    f({
+      id: 'dummyId',
+    });
+    return NO_OP_DISPOSABLE;
+  });
+
+  llamaStackManager.init();
+
+  expect(llamaStackManager.refreshLlamaStackContainer).toHaveBeenCalledWith('dummyId');
+
+  await vi.waitFor(() => {
+    expect(llamaStackManager.getContainerInfo()).toEqual({
+      containerId: 'dummyId',
+      port: 50000,
+    });
+  });
+});
+
+test('onStopContainerEvent event should call refreshLlamaStackContainer and clear containerInfo', async () => {
+  vi.spyOn(llamaStackManager, 'refreshLlamaStackContainer');
+  vi.spyOn(taskRegistry, 'deleteByLabels');
+  vi.mocked(containerEngine.listContainers).mockResolvedValueOnce([LLAMA_STACK_CONTAINER_RUNNING]);
+  vi.mocked(containerRegistry.onStartContainerEvent).mockImplementation(f => {
+    f({
+      id: 'dummyId',
+    });
+    return NO_OP_DISPOSABLE;
+  });
+  vi.mocked(containerRegistry.onStopContainerEvent).mockImplementation(f => {
+    setTimeout(() => {
```

Comment:
same remark as in https://github.com/containers/podman-desktop-extension-ai-lab/pull/2831/files#r2041688355